controllers/subscriptions.go

```diff
@@ -111,6 +111,14 @@ func CheckExistingSubscriptions(cli client.Client, desiredSubscription *operator
 	}
 
 	if !subExsist {
+		// Set the catalog source for the odf-dependencies subscription to match that of the odf-operator subscription
+		// This ensures that the odf-dependencies subscription uses the same catalog source across all environments,
+		// including offline and test environments where the catalog name may vary.
```

Comment:
if this is always the case can we remove loading these specific vars from configmap or env? it's better not to have the data in two places.
(or)
let's take the configmap clubbed with odf-op as source of truth and use that as that'll decouple the hardcoding?

Reply:
I want to have a place where data is present. In 4.19 we will have something concrete and we can rewrite our logic based on that.

Reply:
> I want to have a place where data is present.
- then you already have that from configmap isn't it?

> In 4.19 we will have something concrete and we can rewrite our logic based on that.
- sry, my comment isn't related to logic but of data duplication

Reply:
We had a discussion over a call.